swarm/network/simulation/kademlia_test.go

```diff
@@ -22,13 +22,29 @@ import (
 	"testing"
 	"time"
 
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/node"
 	"github.com/ethereum/go-ethereum/p2p/simulations/adapters"
 	"github.com/ethereum/go-ethereum/swarm/network"
 )
 
+/*
+	TestWaitTillHealthy tests that we indeed get a healthy network after we wait for it.
+	For this to be tested, a bit of a snake tail bite needs to happen:
+		* First we create a first simulation
+		* Run it as nodes connected in a ring
+		* Wait until the network is healthy
+		* Then we create a snapshot
```

Comment:
correct snapshot dump/load is (should be) tested elsewhere.
maybe the only  reason is to 'freeze the network' but then in second sim we switch off discovery.

the  correct way is to create peerpotmap from node addresses of the first sim
and verify health on the snapshot itself.
assuming correct dump/load of snapshot, a quick and dirty way for this last   step is to  create a new control sim with no discovery, load snapshot and check health using peerpotmap from previous sims.
when checking health of each node ,use nodeids extracted from first run.

Reply:
@zelig dump/load is tested in `p2p/simulation:network_test.go`